plutus/budget_manager/mysql.py

```diff
@@ -0,0 +1,162 @@
+    ''' Upserts a row in the budgets mysql table. '''
+
+    budget_dict = protobuf_to_dict(budget, use_enum_labels=True)
+
+    # Construct column values
+    budget_id = budget_dict['name']
+    display_name = budget_dict['display_name']
+
+    # Currently GCP Budgets API limits to either 0 or 1 project assigned to a budget (0 meaning all projects)
+    if ('projects' in budget_dict['budget_filter'] and len(budget_dict['budget_filter']['projects']) == 1):
+        project_number = budget_dict['budget_filter']['projects'][0].replace("projects/", "")
+    else:
+        log.error("Unexpected error. Budgets API currently expects budget_filter.projects to be either 0 or 1 projects.")
+        log.error("Unless this has changed. Also, plutus only currently manages budgets for single projects.")
+        metrics.incr("error_count", tags = ["type:project_budget_filter", f"project_id:{project_id}", f"config_type:{config_type}"])
+        sys.exit(1)
+    
+    if 'services' in budget_dict['budget_filter']:
+        products = ','.join(budget_dict['budget_filter']['services'])
+    else:
+        products = "ALL"
+
+    owner_emails = get_owner_emails_for_project(project_id, alert_emails)
+
+    if budget_dict['amount'].get('specified_amount') is None:
+        budget_type = "LASTMONTH"
+        budget_amount = "-1"
+    else:
+        budget_type = "AMT"
+        budget_amount = budget_dict['amount']['specified_amount']['units']
+
+    if budget_dict['budget_filter']['credit_types_treatment'] == 'INCLUDE_ALL_CREDITS':
+        include_credits = "TRUE"
+    elif budget_dict['budget_filter']['credit_types_treatment'] == 'EXCLUDE_ALL_CREDITS':
+        include_credits = "FALSE"
+    else:
+        log.error("Unexpected error. Credits_type_treatment should be either of INCLUDE_ALL_CREDITS or EXCLUDE_ALL_CREDITS.")
+        log.error("Google may have changed their budgets API.")
+        include_credits = "FALSE"
+        metrics.incr("error_count", tags = ["type:credits_type_treatment", f"project_id:{project_id}", f"config_type:{config_type}"])
+
+    if budget_dict['all_updates_rule'].get('pubsub_topic') is None:
+        pubsub = "FALSE"
+        pubsub_topic = "NA"
+    else:
+        pubsub = "TRUE"
+        pubsub_topic = budget_dict['all_updates_rule']['pubsub_topic']
+
+    now = datetime.datetime.utcnow()
+    curr_time = now.strftime('%Y-%m-%d %H:%M:%S')
+
+    last_modified = curr_time
+    created_date = curr_time
+
+    sql = f""" INSERT INTO budgets (
+budget_id,
+display_name,
+project_id,
+project_number,
+products,
+budget_type,
+budget_amount,
+include_credits,
+pubsub,
+pubsub_topic,
+owner_emails,
+created_date,
+last_modified,
+config_type
+)
+VALUES
+(
+'{budget_id}',
+'{display_name}',
+'{project_id}',
+'{project_number}',
+'{products}',
+'{budget_type}',
+{budget_amount},
+{include_credits},
+{pubsub},
+'{pubsub_topic}',
+'{owner_emails}',
+'{created_date}',
+'{last_modified}',
```

Comment:
This could just be `NOW()` to use mysql timestamp instead of getting the time above.